File: export_decomp.py
# Export decompiled functions to a file
from ghidra.app.decompiler import DecompInterface
from ghidra.util.task import ConsoleTaskMonitor
import os

output_dir = "/Users/inoki/Builds/srcs/Rosetta2"
program = currentProgram
program_name = program.getName()

# Create output file
output_file = os.path.join(output_dir, f"{program_name}_decomp.c")

decomp_interface = DecompInterface()
decomp_interface.openProgram(program)

monitor = ConsoleTaskMonitor()

with open(output_file, "w") as f:
    f.write(f"// Decompiled code for {program_name}\n")
    f.write(f"// Language: {program.getLanguageID()}\n\n")

    functions = program.getFunctionManager().getFunctions(True)
    for func in functions:
        result = decomp_interface.decompileFunction(func, 0, monitor)
        if result.decompileCompleted():
            decompiled_code = result.getDecompiledFunction().getC()
            f.write(decompiled_code)
            f.write("\n\n")

print(f"Decompiled code written to {output_file}")
decomp_interface.dispose()

File: ExportDecomp.java
// Export decompiled functions to a file
import ghidra.app.decompiler.DecompInterface;
import ghidra.util.task.ConsoleTaskMonitor;
import ghidra.program.model.listing.*;
import java.io.*;

import ghidra.app.script.GhidraScript;

public class ExportDecomp extends GhidraScript {
    @Override
    public void run() throws Exception {
        Program program = currentProgram;
        String programName = program.getName();

        String outputDir = "/Users/inoki/Builds/srcs/Rosetta2";
        File outputFile = new File(outputDir, programName + "_decomp.c");

        DecompInterface decompInterface = new DecompInterface();
        decompInterface.openProgram(program);

        ConsoleTaskMonitor monitor = new ConsoleTaskMonitor();

        try (PrintWriter writer = new PrintWriter(new FileWriter(outputFile))) {
            writer.println("// Decompiled code for " + programName);
            writer.println("// Language: " + program.getLanguageID());
            writer.println();

            for (Function func : program.getFunctionManager().getFunctions(true)) {
                var result = decompInterface.decompileFunction(func, 0, monitor);
                if (result.decompileCompleted()) {
                    writer.println(result.getDecompiledFunction().getC());
                    writer.println();
                }
            }

            println("Decompiled code written to " + outputFile.getAbsolutePath());
        } catch (IOException e) {
            e.printStackTrace();
        }

        decompInterface.dispose();
    }
}
